course_discovery/apps/edx_elasticsearch_dsl_extensions/management/commands/search_index.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from collections import namedtuple
+
+from django.conf import settings
+from django.core.management import CommandError
+from django_elasticsearch_dsl.management.commands.search_index import Command as DjangoESDSLCommand
+from django_elasticsearch_dsl.registries import registry
+from elasticsearch_dsl.connections import get_connection
+
+from course_discovery.apps.core.utils import ElasticsearchUtils
+
+OLD_AND_NEW_INDEX_NAMES = slice(2, 4)
+
+AliasMapper = namedtuple('AliasMapper',
+                         'document registered_index registered_index_name new_index_name alias record_count')
+logger = logging.getLogger(__name__)
+
+
+class Command(DjangoESDSLCommand):
```

Comment:
Wait, what is `search_index` for? I thought this had just been a renamed `update_index` which we stopped doing. Are they two separate commands?

Reply:
Yes, these are different commands now. I have not renamed `search_index` to `update_index`. I didn't even think about it. `search_index` is a command from `django-es-dsl` https://django-elasticsearch-dsl.readthedocs.io/en/latest/management.html#. Here we can create, populate data, drop indexes, etc. The only thing I did was change its behavior to create indexes in our thicky way, i.e. add timestamp and operate from aliases. `update_index` is what it was before, when it is possible to create indexes with a validation check, i.e. execute `sanity-check` feature.
Renaming `search_index` doesn't seem like a good idea to me since this command is very different from what was in haystack-django. But leaving `update_index` with custom functionality was a doable task.

Reply:
OK cool, thanks.